File: frontend/src/api/courtRecords.ts
import client from "./client";
import type { CaseSearchResponse, CourtCase, LienSearchResponse } from "../types";

export async function searchFederalCases(params: {
  party_name: string;
  state?: string;
  case_type?: string;
}): Promise<CaseSearchResponse> {
  const { data } = await client.get<CaseSearchResponse>("/v1/court-records/cases", { params });
  return data;
}

export async function getCaseDetails(caseId: string): Promise<CourtCase> {
  const { data } = await client.get<CourtCase>(`/v1/court-records/cases/${caseId}`);
  return data;
}

export async function searchStateLiens(params: {
  debtor_name: string;
  state: string;
  lien_type?: string;
}): Promise<LienSearchResponse> {
  const { data } = await client.get<LienSearchResponse>("/v1/court-records/liens", { params });
  return data;
}

File: src/aloha/api/schemas/court_records.py
"""Pydantic schemas for court records and lien API responses."""

from __future__ import annotations

from pydantic import BaseModel, Field

# ── Cases ─────────────────────────────────────────────────────────────────────


class PartyOut(BaseModel):
    name: str | None = None
    role: str | None = None


class CaseOut(BaseModel):
    case_id: str | None = None
    case_title: str | None = None
    court: str | None = None
    case_type: str | None = None
    filing_date: str | None = None
    status: str | None = None
    parties: list[PartyOut] = Field(default_factory=list)
    docket_url: str | None = None


class CaseDetailResponse(CaseOut):
    error: str | None = None


class CaseSearchResponse(BaseModel):
    cases: list[CaseOut] = Field(default_factory=list)
    error: str | None = None


# ── Liens ─────────────────────────────────────────────────────────────────────


class LienOut(BaseModel):
    filing_number: str | None = None
    debtor: str | None = None
    creditor: str | None = None
    amount: float | None = None
    filing_date: str | None = None
    lien_type: str | None = None
    state: str | None = None


class LienSearchResponse(BaseModel):
    liens: list[LienOut] = Field(default_factory=list)
    error: str | None = None
